oauth.go

```diff
@@ -0,0 +1,41 @@
+package pluginapi
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+// UserService exposes methods to manipulate OAuth Apps.
+type OAuthService struct {
+	api plugin.API
+}
+
+// Create creates a new OAuth App,
+//
+// Minimum server version: 5.38
+func (o *OAuthService) Create(app *model.OAuthApp) error {
+	createdApp, appErr := o.api.CreateOAuthApp(app)
+	if appErr != nil {
+		return normalizeAppErr(appErr)
+	}
+
+	*app = *createdApp
```

Comment:
It seems odd to me that we are mutating the passed in `app`, instead of returning `createdApp`. Or this may be something common in this library?

Reply:
That is in deed common in this library. It allows for shorter code like:
```go
post := &model.Post{Message: "abc", ChannelID: "someID"}
err := client.Post.Create(post)
if err != nil {
	panic(err)
}
log.Printf("Post as ID %s", post.Id)
```

Also, the pattern is consistently used in the code base and hence in this PR.